### PR TITLE
Allow geodataframes in assert_polygons

### PR DESCRIPTION
**Pull request by nkorinek**
Simple fix to allow `GeoDataFrames` to be valid inputs into `assert_polygons`. Also fixed issue with check that input wasn't empty.

### CODE REVIEW

**Comment by nkorinek** [comment]
@lwasser

File: matplotcheck/vector.py [hunk 1]
```diff
@@ -473,14 +473,18 @@ def assert_polygons(
     ):
         """Asserts the polygon data in Axes ax is equal to polygons_expected to decimal place dec with error message m
 		If polygons_expected is am empty list or None, assertion is passed
-		
+
 		Parameters
 		----------
 		polygons_expected: list of polygons expected to be founds on Axes ax
```
Comment (lwasser) [inline matplotcheck/vector.py]: @nkorinek please make sure each input parameter uses numpy style docs. so this should have the expected input type and then on the second line, the explanation of what that object is / does. please follow earthpy so we may need to update many of the docstrings.

**Review by lwasser — CHANGES_REQUESTED** [review]
@nkorinek  great start!! given you are working on the vector module. let's start working on tests for it too. So when you submit new functionality please also include tests for the method that you worked on. i'll wait to merge this until you have added some tests. also please update the change log like we do in earthpy each time.

**Comment by nkorinek** [comment]
@lwasser added changes requested!

**Comment by nkorinek** [comment]
Build is failing due to issues with descartes, we may have to install descartes in the travis environment? Not sure if that's the issue or something is actually wrong with the code itself. @lwasser

**Comment by lwasser** [comment]
ahhh @nkorinek i remember running into this before. we could potentially add descartes as a requirement like geopandas does: 

https://github.com/geopandas/geopandas/blob/2d6b78b28bfa8a4457da81d3b8507309bdf91850/requirements.test.txt#L5 

just for testing purposes. and then add language about installing it IF plotting polygons similar to what geopandas has?? This might help us move forward now and then consider dependencies more carefully in the future?

**Comment by nkorinek** [comment]
@lwasser sounds good! I'll try and be more careful with adding dependencies, I've tried to add a few today!

**Comment by nkorinek** [comment]
@lwasser finally passing all tests!

**Review by lwasser — APPROVED** [review]

**Pull request merged**